listeners.py

```diff
@@ -147,7 +147,25 @@ def process_result(self, obj):
             filepath = self.get_logfile(obj.worker_name)
             self.fds[obj.worker_id] = open(filepath, 'w')
         fd = self.fds[obj.worker_id]
-        fd.write(obj.output)
+
+        # Mark chunks without newlines at the end.
+        #
+        # In OutputWatcher() such chunks are bufferized and
+        # written to the terminal when a newline arrives (to
+        # don't mix partial lines from different workers).
+        #
+        # Here we want to dump output without any buffering,
+        # because it may help with debugging a problem. So
+        # we just mark such chunks and write them out.
+        output = obj.output
+        if not decolor(output).endswith('\n'):
+            output = output + ' <no line carry>\n'
```

Comment:
@Mons proposed to change it to something with a 'newline' word (referring diff's `\ No newline at end of file`). I wrote it as 'line carry' to don't make confusion around a 'new line' and a 'newline symbol', but it seems I introduced an association with the 'line carriage' symbol this way (while we're about the 'line feed' symbol in fact).

The ideal solution (restricting ourself with ASCII) would be something like `<no newline at end of chunk>`, but since it is not a primary information, it should be short. I'll pick up `<no newline>` and will hope that it is more intuitive than `<no line carry>`.

Reply:
Done.